### PR TITLE
Added dynamic label to project multi-select form input

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

* If no projects selected, it reads "(unassigned)"
Screen Shot 2019-04-22 at 4 29 18 PM
* If one project selected, it shows the name of the project
Screen Shot 2019-04-22 at 4 29 25 PM
* If multiple projects selected, it says "n projects"
Screen Shot 2019-04-22 at 4 29 32 PM

### :athletic_shoe: Demo Script / Repro Steps

* Build UI and navigate to https://a2-dev.test/component_library/projects-dropdown

Try selecting the various permutations of projects.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts [hunk 1]
```diff
@@ -56,20 +64,44 @@ export class ProjectsDropdownComponent implements OnInit {
...
+    this.updateProjectsSelected(this.projects[index], checked);
   }
 
   checkboxKeyPress(event, checked, index) {
-    this.projects[index].enabled = !checked;
+    this.updateProjectsSelected(this.projects[index], !checked);
     event.target.setAttribute('checked', !checked);
   }
 
+  updateProjectsSelected(project: Project, enabled: boolean): void {
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts]: `updateSelectedProjects` reads a bit better. i might even simplify to `updateLabel`

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html [hunk 2]
```diff
@@ -9,7 +9,8 @@
     [disabled]="disabled"
     (keydown.arrowup)="moveFocus($event)"
     (keydown.arrowdown)="moveFocus($event)">
-    Select projects... <chef-icon aria-hidden>keyboard_arrow_down</chef-icon>
+    <div id="projects-selected">{{ label }}</div>
+    <chef-icon aria-hidden>keyboard_arrow_down</chef-icon>
   </button>
   <chef-click-outside omit="column-dropdown-toggle" (clickOutside)="closeColumnDropdown()">
```
Comment (msorens) [inline components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html]: Cut-and-paste residual: 
1. Suggest rename to just `closeDropdown` -- there is no "column" lurking around here.
2. Remove the omit property entirely; it is referencing a class that does not exist.
```suggestion
  <chef-click-outside (clickOutside)="closeDropdown()">
```

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts [hunk 3]
```diff
@@ -26,15 +27,22 @@ export class ProjectsDropdownComponent implements OnInit {
   @Input() disabled = false;
 
   // Emits the array of projects to the parent component
-  @Output() onSelection = new EventEmitter<any>();
+  @Output() onSelection = new EventEmitter<Array<Project>>();
+
+  active = false;
+  showError = false;
```
Comment (msorens) [inline components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts]: I do not think `showError` and its associated template code are necessary. (And I was not able to trigger it, anyway.) But the point of it, per the associated error message, is that it requires an item to be selected, and that is not true.

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html [hunk 4]
```diff
@@ -9,7 +9,8 @@
     [disabled]="disabled"
     (keydown.arrowup)="moveFocus($event)"
     (keydown.arrowdown)="moveFocus($event)">
-    Select projects... <chef-icon aria-hidden>keyboard_arrow_down</chef-icon>
+    <div id="projects-selected">{{ label }}</div>
```
[the remark this answers is not in the document]
Comment (susanev) [reply components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html]: Yah, that was the original intention along with a native tooltip on hover to reveal the entire name. @tylercloke would that be something that's quick for you to add now? if not, we can fix this up when @magwalk finishes up the a11y parts of it.

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts (hunk 3 above)
Comment (susanev) [reply components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts]: this component is going to be used in the future for required situations, in which we will need the error.

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (tylercloke) [reply components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html]: I'll commit what micheal has.

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts (hunk 3 above)
Comment (msorens) [reply components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts]: Makes sense. I withdraw my comment then. 😁

File: components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (susanev) [reply components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html]: @tylercloke oh, then i think the user won't be able to access the full project name; i think if we are going to truncate we need to provide the full name in the tool tip, they kinda have to go together.

**Pull request merged**